### PR TITLE
(Ozone+XMB) Savestate thumbnail fullscreen + dropdown

### PR DESCRIPTION
**Pull request by sonninnos**
## Description

This mainly brings these changes to savestate thumbnails for both menu drivers:
- Fullscreen toggling with "left" button
- Show active thumbnail in slot dropdown/submenu 

But it also unifies when regular playlist thumbnails (boxart etc) are shown, since currently
they are rather all over the place, and since they were closely related to the matters at hand.

Best shown with a table in order of menu "clicks", checkmark being when boxart is shown:

||> Playlist| > QM non-running| > QM running|
|---|:---:|:---:|:---:|
|XMB now| :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
|Ozone now| :heavy_check_mark: | |
|Both in this PR| :heavy_check_mark: | :heavy_check_mark: | |

In a nutshell: Don't show playlist thumbnails in Quick Menu while the content is running, but do show when content is not running.

retroarch_2022_06_30_15_10_22_528
retroarch_2022_06_30_15_10_36_343

Other changes in Ozone:
- Fixed image previews under "Load Content"
- Decreased fullscreen thumbnail padding slightly

And in XMB:
- Changed fullscreen image border to "dark" instead of "legacy red" when using "Plain" menu theme

## Reviewers

@jdgleaver 
@bslenul 
@Ryunam 

I tried to be thorough, but helping/testing hands are always appreciated!

### CODE REVIEW

**Comment by sonninnos** [comment]
Issues discovered, so no merging yet.

**Comment by i30817** [comment]
If you'd like some suggestions for a feature, the playlist thumbnail fullscreen view could do with a way to cycle up or down without first having to get out of the view.

**Comment by Ryunam** [comment]
This is amazing and a much appreciated feature! 

I did a few tests using only the Ozone interface and took note of a few visual nitpicks... 

1. When selecting a playlist entry that has a thumbnail in the sidebar and then opening the Quick Menu, the entry thumbnails seem to be reloaded from scratch, with a noticeable and slightly jarring transition, as if the image was _blinking_. I wonder if, when selecting any playlist entry and entering the Quick Menu, it would be feasible to present whatever sidebar thumbnails had been loaded before and avoid this pause/loading.

2. There is a rather common use-case where, if you have some content running, you may pause your content by pressing F1 and entering the RA Menu, then navigate to another playlist entry and access the Quick Menu to quickly boot that new entry, without necessarily closing the piece of content that you were previously running. 
What happens though is that with this PR, when you do the above, the new entry that you are selecting will not show its thumbnail sidebar in the Quick Menu, since the previous content is still running in the background. 

Ideally, when having some content running in the background, the Quick Menu should behave like this:

- not show the thumbnail sidebar if the playlist entry you are selecting matches the content you are currently running;
- show the thumbnail sidebar if you are selecting an entirely different piece of content.

3. This might not be within the scope of this PR, but I feel it's a usability concern that's worth bringing up. Currently, when displaying the fullscreen thumbnails, any input you press will still be detected as normal in the UI and used to control whatever is shown in the background. 
What this means for example is that, when the thumbnail sidebar is displayed in the Quick Menu, pressing Enter will call the fullscreen thumbnail and display the picture(s), but pressing Z with the fullscreen thumbnail visible will activate whichever option was still highlighted in the background menu. This is an issue, because the fullscreen preview of the thumbnail ends up typically covering most of the UI entries in the background, so if all inputs are still being detected you may end up selecting an option (like Remove) that you did not want to select.

Ideally, when showing Fullscreen Thumbnails, only a limited set of controls should be detected (such as pressing Start to enable/disable the Fullscreen preview) and everything else should be blocked.

**Comment by sonninnos** [comment]
Yep, I tried already solving 1+2 beforehand, but did not find a suitable way without causing other issues.

And 3 + 4 seem to be indeed out of this scope and should be addressed later.

**Comment by Ryunam** [comment]
Here's another idea about point 2. I'm thinking that, instead of having to show/hide the sidebar in the Quick Menu depending on whether we have some content running, we could keep a permanent sidebar in the Quick Menu that shows:

- boxart/thumbnail/metadata when highlighting any menu option that is not savestate-related;
- savestate thumbnails only when selecting one of the savestate menu entries.

This would solve point 2, I think, because at that point you wouldn't have to check whether the running content matches the playlist entry that we're selecting to decide whether you need to show the sidebar. The sidebar itself would always be active and simply show different things depending on what's needed. It also makes sense in general to still show the boxart if you are currently running something, as it would act as a (somewhat silly, but still visually coherent) reminder of what you're currently playing.

**Comment by sonninnos** [comment]
Permanent would be too cramped with smaller resolutions, and that cluttering of the running Quick Menu I was specifically trying to avoid..

**Comment by Ryunam** [comment]
Hmm... I see. The issue though is that, once fullscreen thumbnails are extended to the Quick Menu, this feature can be problematic. 

It was fine to have Confirm/Cancel passthrough when this was restricted to the playlists navigation only. With this PR, when you bring up the fullscreen thumbnails you might be already inside the Quick Menu and end up inadvertently selecting some undesired option, which I don't think was being taken into account when that definition was written.

**Comment by sonninnos** [comment]
Oh right. That passthrough is no longer allowed in Quick Menu and State Slot.

**Comment by sonninnos** [comment]
I managed to deal with the thumbnail blink (1).

**Comment by LibretroAdmin** [comment]
Tested the PR, very good. 

The suggestions mentioned here are all good feature suggestions for a followup PR, but probably outside the scope of this PR.

**Comment by Cthulhu-throwaway** [comment]
@sonninnos Couple of warnings added to the Linux build from this PR.

```
menu/drivers/xmb.c:3978:44: warning: ‘%s’ directive output may be truncated writing up to 254 bytes into a region of size 63 [-Wformat-truncation=]
menu/drivers/ozone.c:6395:44: warning: ‘%s’ directive output may be truncated writing up to 254 bytes into a region of size 63 [-Wformat-truncation=]
```
